core/src/main/java/google/registry/batch/RefreshDnsOnHostRenameAction.java

```diff
@@ -206,7 +207,9 @@ public final void map(@Nullable final DomainBase domain) {
       Key<HostResource> referencingHostKey = null;
       for (DnsRefreshRequest request : refreshRequests) {
         if (isActive(domain, request.lastUpdateTime())
-            && domain.getNameservers().contains(request.hostKey())) {
+            && domain
+                .getNameservers()
+                .contains(VKey.create(HostResource.class, request.hostKey()))) {
```

Comment:
What is VKey accomplishing for us here?  Ultimately the way this should work is that there is an array of repoIds on the domain, all of which are foreign keys to the hosts table which has the repoId as its id.  What is VKey getting us?

Reply:
I borrowed this idea when I was refactoring the `TransactionManager` to create a unified API but I made a little bit change to allow `VKey` to be able to carry both ofy key and sql key. Then, when `tm().load(VKey.create(clazz, ofyKey, sqlKey))` is invoked, `JpaTransactionManger` would use `vkey.getSqlKey()` while `DatastoreTransactionManager` can use `vkey.getOfyKey()` to get the key it needs.